### PR TITLE
Keys Sync implementation

### PR DESCRIPTION
**Pull request by maririos**
It depends on PR https://github.com/Azure/azure-sdk-for-net/pull/6479 so while that one gets merged, marking this one as draft.

Actual changes for this PR are in commit https://github.com/Azure/azure-sdk-for-net/commit/5850d36c68ea5e1401dd4c190a65597b279ccd05

------------------------------------
**Update**: PR has been updated to include latest changes in the Keys API.
It is ready to review

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs [hunk 1]
```diff
@@ -43,7 +43,12 @@ public KeyClient(Uri vaultUri, TokenCredential credential, KeyClientOptions opti
 
         public virtual Response<Key> CreateKey(string name, KeyType keyType, KeyCreateOptions keyOptions = default, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            if (string.IsNullOrEmpty(name)) throw new ArgumentException($"{nameof(name)} can't be empty or null");
+            if (keyType == default) throw new ArgumentNullException(nameof(keyType));
+
+            var parameters = new KeyRequestParameters(keyType, keyOptions);
+
+            return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(name), cancellationToken, KeysPath, name);
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs]: >return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(name), cancellationToken, KeysPath, name); [](start = 12, length = 111)

I missed this in the async review, but the path for create is actually /keys/{key-name}/create

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs [hunk 2]
```diff
@@ -58,7 +63,11 @@ public virtual async Task<Response<Key>> CreateKeyAsync(string name, KeyType key
 
         public virtual Response<Key> CreateEcKey(EcKeyCreateOptions ecKey, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            if (ecKey == default) throw new ArgumentNullException(nameof(ecKey));
+
+            var parameters = new KeyRequestParameters(ecKey);
+
+            return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(ecKey.Name), cancellationToken, KeysPath, ecKey.Name);
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs]: >return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(ecKey.Name), cancellationToken, KeysPath, ecKey.Name); [](start = 12, length = 123)

Same comment as above

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs [hunk 3]
```diff
@@ -72,7 +81,11 @@ public virtual async Task<Response<Key>> CreateEcKeyAsync(EcKeyCreateOptions ecK
 
         public virtual Response<Key> CreateRsaKey(RsaKeyCreateOptions rsaKey, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            if (rsaKey == default) throw new ArgumentNullException(nameof(rsaKey));
+
+            var parameters = new KeyRequestParameters(rsaKey);
+
+            return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(rsaKey.Name), cancellationToken, KeysPath, rsaKey.Name);
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs]: >return SendRequest(HttpPipelineMethod.Put, parameters, () => new Key(rsaKey.Name), cancellationToken, KeysPath, rsaKey.Name); [](start = 12, length = 125)

Same comment as above

**Review by schaabs — APPROVED** [review]
:shipit:

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs (hunk 1 above)
Comment (pakrym) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs]: Are you sure? https://docs.microsoft.com/en-us/rest/api/keyvault/setsecret/setsecret

Comment (maririos) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs]: Thank you! I fixed it with the versions url too

**Pull request merged**